### DISCUSSION TITLE
Generic approach to billboards

### DISCUSSION THREAD

**Discussion by tantilla** [donmccurdy/glTF-Transform#1102] [Q&A] [answered]
Hi all,

Just getting started learning glTF 2.0.
From what I can tell so far there is no standard approach to storing/loading billboards.
Apparently 1.0 supported shaders, but no longer. I've seen some discussions involving annotations.
Has anyone in the community found an approach or maybe an extension that works nicely across different runtimes?

Thanks.

**Comment by donmccurdy** [comment, marked as answer]
Hi @tantilla — While there are a number of tools that do features similar to billboards (model-viewer comes to mind: https://modelviewer.dev/examples/annotations/index.html) based on glTF, I think you're correct that there is no standard approach to representing that data in a glTF file. I'm not aware of a standard way to transfer billboard data in other file formats either, unfortunately.

glTF Transform could certainly be used to add billboards to files with a [custom extension](https://gltf-transform.dev/extensions), or to optimize files already using a custom extension, but the work of implementing the billboard support in whatever renderer you're using would remain.

It might be possible to get more answers in the Khronos Slack server, Discord server, or perhaps the main glTF Github repository.

**Comment by tantilla** [comment]
Thanks for your suggestions @donmccurdy.
This model viewer looks fantastic !
I may be able to remove all my Three dependencies and just use glTf.
